### PR TITLE
Adjust to smoldot version 0.2.0

### PR DESCRIPTION
**Pull request by wirednkod**

### CODE REVIEW

**Review by raoulmillais — APPROVED** [review]

File: packages/smoldot-test-utils/src/index.ts [hunk 1]
```diff
@@ -88,8 +90,9 @@ export const mockSmoldot = (responder: RpcResponder, healthResponder = healthyRe
       return Promise.resolve({
         terminate: () => {},
         // fake the async reply by using the reponder to format
-        // a reply via options.json_rpc_callback
-        send_json_rpc: fakeRpcSend(options, responder, healthResponder)
+        // a reply via options.jsonRpcCallback
+        sendJsonRpc: fakeRpcSend(options, responder, healthResponder),
```
Comment (tomaka) [inline packages/smoldot-test-utils/src/index.ts:94]: Misses a parameter?

File: packages/smoldot-test-utils/src/index.ts [hunk 2]
```diff
@@ -102,12 +105,13 @@ export const smoldotSpy = (responder: RpcResponder, rpcSpy: any, healthResponder
     start: async (options: SmoldotOptions): Promise<SmoldotClient> => {
       return Promise.resolve({
         terminate: () => {},
-        send_json_rpc: (rpc: string) => {
+        sendJsonRpc: (rpc: string) => {
```
Comment (tomaka) [inline packages/smoldot-test-utils/src/index.ts:108]: Same

Comment (wirednkod) [reply packages/smoldot-test-utils/src/index.ts:108]: These are mocks - not the actual ones

File: packages/smoldot-test-utils/src/index.ts (hunk 1 above)
Comment (wirednkod) [reply packages/smoldot-test-utils/src/index.ts:94]: These are mocks - not the actual ones

Comment (raoulmillais) [reply packages/smoldot-test-utils/src/index.ts:94]: The parameter is actually missing on line 67. `fakeRpcSend` return a function - that returned function needs the new parameter

**Review by raoulmillais — CHANGES_REQUESTED** [review]

**Comment by wirednkod** [comment]
@raoulmillais that is strange - test do not seem to complain - but are passing

File: packages/smoldot-test-utils/src/index.ts (hunk 2 above)
Comment (tomaka) [reply packages/smoldot-test-utils/src/index.ts:108]: Even if they're mock, they're still missing a parameter?

```suggestion
        sendJsonRpc: (rpc: string, chainIndex: number) => {
```

File: packages/smoldot-test-utils/src/index.ts [hunk 3]
```diff
@@ -88,8 +90,9 @@ export const mockSmoldot = (responder: RpcResponder, healthResponder = healthyRe
       return Promise.resolve({
         terminate: () => {},
         // fake the async reply by using the reponder to format
-        // a reply via options.json_rpc_callback
-        send_json_rpc: fakeRpcSend(options, responder, healthResponder)
+        // a reply via options.jsonRpcCallback
+        sendJsonRpc: fakeRpcSend(options, responder, healthResponder, 0),
```
Comment (tomaka) [inline packages/smoldot-test-utils/src/index.ts:94]:
```suggestion
        sendJsonRpc: fakeRpcSend(options, responder, healthResponder),
```

File: packages/smoldot-test-utils/src/index.ts [hunk 4]
```diff
@@ -53,27 +55,27 @@ export const devChainHealthResponder = (requestJSON: string) => {
...
+// `jsonRpcCallback` after it is called which returns the response
 // returned by the supplied `responder`.
 //
 // If this is called with an rpc string containing the word subscribe (but not
 // unsubscribe), it is considered a subscription and `responder` must have 2
 // mock responses for each subscriptionrequest that the test will make.  The
 // first returning the subscription id.  The second a response to the
 // subscription.
-const fakeRpcSend = (options: SmoldotOptions, responder: RpcResponder, healthResponder: RpcResponder) => {
+const fakeRpcSend = (options: SmoldotOptions, responder: RpcResponder, healthResponder: RpcResponder, chainIndex: number) => {
```
Comment (tomaka) [inline packages/smoldot-test-utils/src/index.ts:66]:
```suggestion
const fakeRpcSend = (options: SmoldotOptions, responder: RpcResponder, healthResponder: RpcResponder) => {
```

File: packages/smoldot-test-utils/src/index.ts [hunk 5]
```diff
@@ -53,27 +55,27 @@ export const devChainHealthResponder = (requestJSON: string) => {
...
   return (rpcRequest: string) => {
```
Comment (tomaka) [inline packages/smoldot-test-utils/src/index.ts:67]:
```suggestion
  return (rpcRequest: string, chainIndex: number) => {
```

File: packages/smoldot-test-utils/src/index.ts [hunk 6]
```diff
@@ -102,12 +105,13 @@ export const smoldotSpy = (responder: RpcResponder, rpcSpy: any, healthResponder
...
       return Promise.resolve({
         terminate: () => {},
-        send_json_rpc: (rpc: string) => {
+        sendJsonRpc: (rpc: string) => {
           // record the message call
           rpcSpy(rpc);
           // fake the async reply using the responder
-          fakeRpcSend(options, responder, healthResponder)(rpc);
-        }
+          fakeRpcSend(options, responder, healthResponder, 0)(rpc)
```
Comment (tomaka) [inline packages/smoldot-test-utils/src/index.ts:112]:
```suggestion
          fakeRpcSend(options, responder, healthResponder)(rpc, chainIndex)
```

**Review by tomaka — CHANGES_REQUESTED** [review]

File: packages/smoldot-test-utils/src/index.ts (hunk 3 above)
Comment (wirednkod) [reply packages/smoldot-test-utils/src/index.ts:94]: yeap - Fixing it now

File: packages/smoldot-test-utils/src/index.ts (hunk 4 above)
Comment (wirednkod) [reply packages/smoldot-test-utils/src/index.ts:66]: yeap - fixing it now

File: packages/smoldot-test-utils/src/index.ts (hunk 5 above)
Comment (wirednkod) [reply packages/smoldot-test-utils/src/index.ts:67]: yeap fixing it now

File: packages/smoldot-test-utils/src/index.ts (hunk 6 above)
Comment (wirednkod) [reply packages/smoldot-test-utils/src/index.ts:112]: yeap fixing it now

**Review by tomaka — APPROVED** [review]

**Pull request merged**